### ISSUE TITLE
pyre error in network time sync

### ISSUE THREAD

**Issue by mkassner** [pupil-labs/pupil#749]
A user reported the following:

```shell
user:~$ pupil_capture 
MainProcess - [INFO] os_utils: Disabling idle sleep not supported on this OS version.
world - [INFO] world: Application Version: 0.9.12.2
world - [INFO] world: System Info: User: testing, Platform: Linux, Machine: user, Release: 4.8.0-54-generic, Version: #57~16.04.1-Ubuntu SMP Wed May 24 16:22:28 UTC 2017
world - [INFO] plugin: Loaded: UVC_Source
world - [INFO] plugin: Loaded: Pupil_Data_Relay
world - [INFO] plugin: Loaded: Pupil_Remote
world - [INFO] plugin: Loaded: Dummy_Gaze_Mapper
world - [INFO] plugin: Loaded: Log_Display
world - [INFO] plugin: Loaded: UVC_Manager
world - [INFO] plugin: Loaded: Screen_Marker_Calibration
world - [DEBUG] network_time_sync: Timer Server ready on port: 39151
Exception in thread Thread-8:
Traceback (most recent call last):
  File "pyre/zbeacon.py", line 118, in prepare_udp
OSError: [Errno 19] No such device
```

**Comment by mkassner** [comment]
Can this be reproduced when turning of all networking on Ubuntu?

**Comment by papr** [comment]
This error is actually caught by `pyre`. The actual reason for the crash is that `pyre` uses `logger.exception()` which automatically adds exception information to the log record dictionary. Our `ZMQ_handler` logger fails when serializing the exception class:

```
Traceback (most recent call last):
...
  File "/Users/pabloprietz/.virtualenvs/py3.6/lib/python3.6/site-packages/pyre/zbeacon.py", line 132, in prepare_udp
    logger.exception("Initializing of {0} raised an exception".format(self.__class__.__name__))
 ...
    self.emit(record)
  File "/Users/pabloprietz/work/pupil/pupil_src/shared_modules/zmq_tools.py", line 37, in emit
    record.__dict__)
  File "/Users/pabloprietz/work/pupil/pupil_src/shared_modules/zmq_tools.py", line 125, in send
    serialized_payload = serializer.dumps(payload, use_bin_type=True)
...
TypeError: can't serialize <class 'OSError'>
```

**Comment by papr** [comment]
A possible fix could look like this:
```diff
diff --git a/pupil_src/shared_modules/zmq_tools.py b/pupil_src/shared_modules/zmq_tools.py
index 4c86f63b..50052c0f 100644
--- a/pupil_src/shared_modules/zmq_tools.py
+++ b/pupil_src/shared_modules/zmq_tools.py
@@ -33,8 +33,12 @@ class ZMQ_handler(logging.Handler):
         self.socket = Msg_Dispatcher(ctx, ipc_pub_url)

     def emit(self, record):
-        self.socket.send('logging.{0}'.format(record.levelname.lower()),
-                         record.__dict__)
+        try:
+            self.socket.send('logging.{0}'.format(record.levelname.lower()), record.__dict__)
+        except TypeError:
```

**Comment by willpatera** [comment]
@mkassner @papr is this still an issue?

**Comment by papr** [comment]
This has been fixed with 46c08e8397eec3b5caaa1d1a10fc5e107f4cdf7b
